Problem compiling Snappyer in v2.2.1 in Mac OS X

Hello, I'm trying to update brod to v2.2.1 and I've got error when trying to compile snappyer on Mac OS X (El Capitan). I'm using Hex in an Elixir project.

A co-worker that also uses Mac OS X has the same problem. Meanwhile, it seems that compiles perfectly in another co-worker's machine that uses Ubuntu (but our acceptance tests stop working with message parse errors ...)

With version 2.1.8 everything works perfectly.

This is the output that I got:

_==> snappyer (compile)
 DEPEND snappyer.d
 ERLC   snappyer.erl
 APP    snappyer.app.src
 CPP    snappy-sinksource.cc
clang: warning: optimization flag '-finline-functions' is not supported
 CPP    snappy-stubs-internal.cc
clang: warning: optimization flag '-finline-functions' is not supported
 CPP    snappy.cc
clang: warning: optimization flag '-finline-functions' is not supported
 CPP    snappyer.cc
clang: warning: optimization flag '-finline-functions' is not supported
 LD     snappyer.so
Undefined symbols for architecture x86_64:
  "_enif_alloc_binary", referenced from:
      SnappyNifSink::SnappyNifSink(enif_environment_t*) in snappyer.o
      SnappyNifSink::SnappyNifSink(enif_environment_t*) in snappyer.o
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
  "_enif_inspect_iolist_as_binary", referenced from:
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
      _snappy_uncompressed_length_erl in snappyer.o
      _snappy_is_valid in snappyer.o
  "_enif_make_atom", referenced from:
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
      _snappy_uncompressed_length_erl in snappyer.o
      _snappy_is_valid in snappyer.o
  "_enif_make_badarg", referenced from:
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
      _snappy_uncompressed_length_erl in snappyer.o
      _snappy_is_valid in snappyer.o
  "_enif_make_binary", referenced from:
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
  "_enif_make_existing_atom", referenced from:
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
      _snappy_uncompressed_length_erl in snappyer.o
      _snappy_is_valid in snappyer.o
  "_enif_make_tuple", referenced from:
      _snappy_compress_erl in snappyer.o
      _snappy_decompress_erl in snappyer.o
      _snappy_uncompressed_length_erl in snappyer.o
      _snappy_is_valid in snappyer.o
  "_enif_make_ulong", referenced from:
      _snappy_uncompressed_length_erl in snappyer.o
  "_enif_realloc_binary", referenced from:
      SnappyNifSink::GetAppendBuffer(unsigned long, char*) in snappyer.o
      SnappyNifSink::getBin() in snappyer.o
      _snappy_compress_erl in snappyer.o
  "_enif_release_binary", referenced from:
      SnappyNifSink::~SnappyNifSink() in snappyer.o
      SnappyNifSink::~SnappyNifSink() in snappyer.o
      SnappyNifSink::~SnappyNifSink() in snappyer.o
      _snappy_compress_erl in snappyer.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/samuel/cfadapter/deps/snappyer/priv/snappyer.so] Error 1
ERROR: Command [compile] failed!
==> cfadapter
** (Mix) Could not compile dependency :snappyer, "/Users/samuel/.mix/rebar compile skip_deps=true deps_dir="/Users/samuel/cfadapter/_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile snappyer", update it with "mix deps.update snappyer" or clean it with "mix deps.clean snappyer"_

I'm not used to NIFs in Erlang/Elixir, so I'm not sure where is the problem. Any idea???

Thank you.

--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,11 @@
 PROJECT = brod
 PROJECT_DESCRIPTION = Kafka client library in Erlang
-PROJECT_VERSION = 2.2.1
+PROJECT_VERSION = 2.2.2
 
 DEPS = supervisor3 kafka_protocol
 
 dep_supervisor3_commit = 1.1.3
-dep_kafka_protocol_commit = 0.7.2
+dep_kafka_protocol_commit = 0.7.3
 
 TEST_DEPS = meck proper
 

--- a/rebar.config
+++ b/rebar.config
@@ -1,4 +1,4 @@
 {deps, [ {supervisor3, "1.1.3"}
-       , {kafka_protocol, "0.7.2"}
+       , {kafka_protocol, "0.7.3"}
        ]}.
 {erl_opts, [warn_unused_vars,warn_shadow_vars,warn_unused_import,warn_obsolete_guard,debug_info]}.

--- a/rebar.lock
+++ b/rebar.lock
@@ -1,10 +1,10 @@
 {"1.1.0",
-[{<<"kafka_protocol">>,{pkg,<<"kafka_protocol">>,<<"0.7.2">>},0},
- {<<"snappyer">>,{pkg,<<"snappyer">>,<<"1.1.3-1.0.3">>},1},
+[{<<"kafka_protocol">>,{pkg,<<"kafka_protocol">>,<<"0.7.3">>},0},
+ {<<"snappyer">>,{pkg,<<"snappyer">>,<<"1.1.3-1.0.4">>},1},
  {<<"supervisor3">>,{pkg,<<"supervisor3">>,<<"1.1.3">>},0}]}.
 [
 {pkg_hash,[
- {<<"kafka_protocol">>, <<"DB7992C8A18DB373057EE9F88554261B7B946A6894D8CBF9BA7C9D3C48AA24E6">>},
- {<<"snappyer">>, <<"964F889E9EFAFBC5B4036EFE257BCB2EAD679BC6CAF72544F759EAECA9C0A36C">>},
+ {<<"kafka_protocol">>, <<"A08844DB51F6D4E172CC1C315862DEE9575EF10B05FE38D17CC3F31CF46AAA4D">>},
+ {<<"snappyer">>, <<"83EC59E7A9627CF5687096383C8BC508C7DB24127CD3CAEA04FE107BA3E47688">>},
  {<<"supervisor3">>, <<"96D0553CA94F7998EC515004CAF0419E0BC989907EBCEBCACA1D3E19AF0D52F6">>}]}
 ].

--- a/src/brod.app.src
+++ b/src/brod.app.src
@@ -1,7 +1,7 @@
 %% -*- mode:erlang -*-
 {application,brod,
  [{description,"Apache Kafka Erlang client library"},
-  {vsn,"2.2.1"},
+  {vsn,"2.2.2"},
   {registered,[]},
   {applications,[kernel,stdlib,supervisor3,kafka_protocol]},
   {env,[]},

